typing/layouts.ml

```diff
@@ -388,23 +394,74 @@ module Layout = struct
   module Violation = struct
     open Format
 
+    let printtyp_path = ref (fun _ _ -> assert false)
+
+    let set_printtyp_path f = printtyp_path := f
```

Comment:
Not related to your patch, but this makes me realize that we should just play a similar game with `type_expr` and simplify the types of `report_with_offender`.  That function is always passed `Printtyp.type_expr` - I thought I would end up passing other things, but in the end there's no reason it needs its (pretty confusing) type.  Maybe leave a CR to consider that change, while you're here.